src/Icicle/Avalanche/Statement/Simp.hs

```diff
@@ -33,6 +36,56 @@ import qualified    Data.List as List
 import              Control.Monad.Trans.Class
 
 
+convertValues :: a -> Statement a n Prim -> Statement a n Prim
+convertValues a_fresh statements
+ = runIdentity
+ $ transformUDStmt trans () statements
+ where
+  trans _ s
```

Comment:
I think there is a "transformX" combinator which would be ever-so-slightly easier